src/completePostAnalysis.ts

```
@@ -0,0 +1,135 @@
+import type OpenAI from 'openai';
+import openai from './openaiClient';
+import { categorizePost, type Categorization } from './classifyWithOpenAI';
+import { analyzeMultiplePosts, type SentimentResult } from './analyzeSentiment';
+import { generateTitleForPost, type TitleResult } from './generateTitle';
+
+// Combined result type for complete post analysis
+export type PostAnalysisResult = {
+    post: string;
+    title?: string;
+    categorization?: Categorization;
+    sentiment?: "BULLISH" | "BEARISH" | "NEUTRAL";
+    errors?: string[];
+};
+
+// Process a single post with all analysis
+export async function analyzeCompletePost(
+    post: string,
+    clientOverride?: OpenAI
+): Promise<PostAnalysisResult> {
+    const usedClient: OpenAI = clientOverride ?? openai;
+    const result: PostAnalysisResult = { post, errors: [] };
+
+    // Generate title
+    try {
+        const title = await generateTitleForPost(post, usedClient);
+        if (title) {
+            result.title = title;
+        }
+    } catch (e) {
+        result.errors?.push(`Title generation failed: ${e}`);
+        console.error("Title generation error:", e);
+    }
+
+    // Categorize post
+    try {
+        const categorization = await categorizePost(post, usedClient);
+        if (categorization) {
+            result.categorization = categorization;
+        }
+    } catch (e) {
+        result.errors?.push(`Categorization failed: ${e}`);
+        console.error("Categorization error:", e);
+    }
+
+    // Analyze sentiment
+    try {
+        const sentimentResults = await analyzeMultiplePosts([post]);
+        if (sentimentResults.length > 0) {
+            result.sentiment = sentimentResults[0].sentiment;
+        }
+    } catch (e) {
+        result.errors?.push(`Sentiment analysis failed: ${e}`);
+        console.error("Sentiment analysis error:", e);
+    }
+
+    return result;
```

if errors.length > 0 then we need to throw an error otherwise return result. its either error is thrown or we return all the data properly